File: ros2_python_tutorial/ros2_python_tutorial/p1_publisher.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class MinimalPublisher(Node):
    def __init__(self):
        super().__init__('minimal_publisher')
        self.publisher_fast = self.create_publisher(String, 'fast_topic', 10)
        self.timer_fast = self.create_timer(0.5, self.timer_callback_fast)
        self.publisher_slow = self.create_publisher(String, 'slow_topic', 10)
        self.timer_slow = self.create_timer(1.0, self.timer_callback_slow)
        self.fast_i = 0
        self.slow_i = 0
    def timer_callback_fast(self):
        msg = String()
        msg.data = 'Fast Timer : %d' % self.fast_i
        self.publisher_fast.publish(msg)
        self.get_logger().info('[Fast] Publishing "%s"' % msg.data)
        self.fast_i +=1
    def timer_callback_slow(self):
        msg = String()
        msg.data = 'Slow Timer : %d' % self.slow_i
        self.publisher_slow.publish(msg)
        self.get_logger().info('[Slow] Publishing "%s"' % msg.data)
        self.slow_i +=1
def main(args=None):
    rclpy.init(args=args)
    minimal_publisher = MinimalPublisher()
    rclpy.spin(minimal_publisher)
    minimal_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ros2_cpp_tutorial/src/p1_publisher.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

using namespace std::chrono_literals;

class MinimalPublisher : public rclcpp::Node {
    public:
        MinimalPublisher() : Node("minimal_publisher"), count_fast_(0),count_slow_(0) {
            publisher_fast_ = this->create_publisher<std_msgs::msg::String>("fast_topic", 10);
            timer_fast_ = this-> create_wall_timer(500ms, std::bind(&MinimalPublisher::timer_callback_fast, this));
            publisher_slow_ = this->create_publisher<std_msgs::msg::String>("slow_topic", 10);
            timer_slow_ = this-> create_wall_timer(1000ms, std::bind(&MinimalPublisher::timer_callback_slow, this));

        }
    private:
        void timer_callback_fast() {
            auto message = std_msgs::msg::String();
            message.data = "Fast Timer " + std::to_string(count_fast_++);
            RCLCPP_INFO(this->get_logger(), "[Fast] Publishing: '%s'", message.data.c_str());
            publisher_fast_->publish(message);
        }
        void timer_callback_slow() {
            auto message = std_msgs::msg::String();
            message.data = "Slow Timer " + std::to_string(count_slow_++);
            RCLCPP_INFO(this->get_logger(), "[Slow] Publishing: '%s'", message.data.c_str());
            publisher_slow_->publish(message);        
        }
        rclcpp::TimerBase::SharedPtr timer_fast_;
        rclcpp::TimerBase::SharedPtr timer_slow_;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_fast_;
        rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_slow_;
        size_t count_fast_;
        size_t count_slow_;
};

int main(int argc, char* argv[]) {
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<MinimalPublisher>());
    rclcpp::shutdown();
    return 0;
}